flux-driver/src/collector.rs

```diff
@@ -358,11 +373,35 @@ impl<'tcx, 'a> SpecCollector<'tcx, 'a> {
             ("ignore", AttrArgs::Empty) => FluxAttrKind::Ignore,
             ("opaque", AttrArgs::Empty) => FluxAttrKind::Opaque,
             ("trusted", AttrArgs::Empty) => FluxAttrKind::Trusted,
+            ("extern_spec", AttrArgs::Empty) => FluxAttrKind::ExternSpec,
             _ => return Err(self.emit_err(errors::InvalidAttr { span: attr_item.span() })),
         };
         Ok(FluxAttr { kind, span: attr_item.span() })
     }
 
+    // In Prusti they suggested looking into doing this instead of using a Visitor...
+    // it seems more brittle but I guess conversely their version is a little permissive.
+    fn extract_extern_def_id_from_extern_spec_fn(
+        &mut self,
+        def_id: LocalDefId,
+    ) -> Result<DefId, ErrorGuaranteed> {
+        use rustc_hir::{def, ExprKind, Node};
+        if let Node::Item(i) = self.tcx.hir().find_by_def_id(def_id).unwrap()
+            && let ItemKind::Fn(_, _, body_id) = &i.kind
+            && let Node::Expr(e) = self.tcx.hir().find(body_id.hir_id).unwrap()
+            && let ExprKind::Block(b, _) = e.kind
+            && let Some(e) = b.expr
+            && let ExprKind::Call(callee, _) = e.kind
+            && let ExprKind::Path(ref qself) = callee.kind {
+                let typeck_result = self.tcx.typeck(def_id);
```

Comment:
There's still no consensus on how rustfmt will format this but I've been formatting it like so to be consistent with how rustfmt formats multiline if conditions

```suggestion
            && let ExprKind::Path(ref qself) = callee.kind 
        {
                let typeck_result = self.tcx.typeck(def_id);
```